pandas/core/frame.py

```diff
@@ -6273,6 +6274,10 @@ class    max    type
             else:
                 to_insert = ((self.index, None),)
 
+            if len(new_obj.columns) == 0 and names:
```

Comment:
This seems to be a similar issue like the pattern of creating an empty dataframe and then adding columns that I also encountered in the tests (for now I always got the tests passing by either ensuring the expected uses object dtype or ensuring the empty dataframe starts with an empty columns Index of dtype `"str"`).

I am not sure we should "fix" this issue, as it would also introduce an inconsistency in the expected dtype, but opened https://github.com/pandas-dev/pandas/issues/60338 to give this a bit more visibility.

Reply:
> I am not sure we should "fix" this issue

I think any code changes should perhaps be separate PRs to the general resolving xfails PRs and maybe to avoid any regressions on 2.3.x be wrapped in using_string_dtype if blocks?

Reply:
Thanks for the feedback. I'll get this removed

@simonjayhawkins just to confirm I understand, are you asking to separate our PRs that need to change tests to correct the xfails from PRs that need to change the core implementation?

Reply:
if the PR title implies that the changes are test related then I don't normally expect to see code changes to the core implementation, so yes, I think splitting this PR is wise.